File: benchmarks/cross_runtime/fib.lua
-- Recursao pura: custo de chamada de funcao + aritmetica de inteiros.
local function fib(n)
  if n <= 1 then return n end
  return fib(n - 1) + fib(n - 2)
end

local function main()
  print("CHECKSUM:" .. fib(30))
end

main()

File: benchmarks/cross_runtime/fib.py
# Recursao pura: custo de chamada de funcao + aritmetica de inteiros.
def fib(n: int) -> int:
    if n <= 1:
        return n
    return fib(n - 1) + fib(n - 2)


def main():
    print(f"CHECKSUM:{fib(30)}")


main()
